src/plugins/intel_cpu/tests/unit/shape_inference_test/utils.hpp

```diff
@@ -21,7 +21,12 @@ void shape_inference(ov::Node* op,
                      std::vector<StaticShape>& output_shapes,
                      const std::map<size_t, TTensorPtr>& constant_data = {}) {
     const auto shape_infer = make_shape_inference<TIface>(op->shared_from_this());
-    auto result = shape_infer->infer(input_shapes, constant_data);
+    IShapeInferCommon::Result result;
+    if (auto static_shape_infer = dynamic_cast<IStaticShapeInfer*>(shape_infer.get())) {
```

Comment:
I think check of `shape_inference` type is not required here, this function is template, use `TIface` template parameter on call to specify if new version of inference should be used.

Reply:
I think we currently still use make_shape_inference\<IShapeInferCommon\> from the [code](https://github.com/openvinotoolkit/openvino/blob/ff4ed54bba9d496503645e7dd0e7379336c6b0ae/src/plugins/intel_cpu/src/utils/shape_inference/shape_inference_cpu.cpp#L10), make_shape_inference\<IStaticShapeInfer\> is called by make_shape_inference\<IShapeInferCommon\> at [code](https://github.com/openvinotoolkit/openvino/blob/ff4ed54bba9d496503645e7dd0e7379336c6b0ae/src/plugins/intel_cpu/src/utils/shape_inference/shape_inference.cpp#L714), so maybe it will better still call make_shape_inference\<IShapeInferCommon\> in the test.

Reply:
My comment was not accurate. 

This function is test only function by default use old iface (IShapeInferCommon). All ops which supports new iface (will be returned as IShapeInferCommon) also supports old iface but not vice versa. The tests use old version of interface and cast is not required.

In all tests this helper is used to create old version, but if required it can be forced to use new iface e.g. shape_inference<IStaticShapeInfer>(...).

The point is that internal if statement with cast is not required as selection can be done by template parameter.

When all ops will be migrated the default value of template parameter can be updated or removed at all and all tests will use new iface.

Reply:
thanks your explanation, the main change in this pr is **dynamic cast from IShapeInferCommon to IShapeInferStatic** currently, it is not covered by unit test case, I also don't know how to cover it now. from my opinion,  CPU unit test case should cover 
```
   NgraphShapeInfer::infer(
        const std::vector<std::reference_wrapper<const VectorDims>>& input_shapes,
        const std::unordered_map<size_t, MemoryPtr>& data_dependency) {
```
but currently unit test cover `shape_infer->infer(input_shapes, const_data);`  or `shape_infer->infer(input_shapes, ov::teonsor)`,

Reply:
OK,  this is test code only, and probably will  change soon.